src/robot/libraries/BuiltIn.py

```diff
@@ -1100,6 +1223,10 @@ def should_not_match(self, string, pattern, msg=None, values=True,
         ``*``, ``?`` and ``[chars]`` acting as wildcards. See the
         `Glob patterns` section for more information.
 
+        If ``ignore_case`` is given a true value (see `Boolean arguments`) and
+        compared items are strings, it indicates that comparison should be
+        case-insensitive.
```

Comment:
Great catch that this documentation was missing!

With this keyword arguments are always strings, so the `and compared items are strings.` can be removed. Same applies to the other place where this documentation was added.